### ISSUE TITLE
Query crashes for avg_zero aggregation method

### ISSUE THREAD

**Issue by eachirei** [graphite-project/graphite-web#2337]
What you attempted: Hitcount query
Expected Result: Successful result
Witnessed Result: Error described below
System Details
OS / Version: Docker container from ubuntu:14.04
Python Version: 2.7.6
Django Version: 1.11.10
Graphite Component Versions: 1.1.3

When querying for data sometimes the graphite request returns the following error:
```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/lib/graphite/webapp/graphite/render/views.py", line 117, in renderView
    data.extend(evaluateTarget(requestContext, targets))
  File "/var/lib/graphite/webapp/graphite/render/evaluator.py", line 15, in evaluateTarget
    prefetchData(requestContext, pathExpressions)
  File "/var/lib/graphite/webapp/graphite/render/datalib.py", line 262, in prefetchData
    for result in STORE.fetch(pathExpressions, startTime, endTime, now, requestContext):
  File "/var/lib/graphite/webapp/graphite/storage.py", line 179, in fetch
    'fetch for %s' % str(patterns))
  File "/var/lib/graphite/webapp/graphite/storage.py", line 132, in wait_jobs
    raise Exception(message)
Exception: All requests failed for fetch for [u'stats.request.*.all'] (1)

fetch for [u'stats.request.*.all']: Invalid consolidation function: 'avg_zero': Traceback (most recent call last):

  File "/var/lib/graphite/webapp/graphite/worker_pool/pool.py", line 43, in run
    self.result = self.func(*self.args, **self.kwargs)

  File "/var/lib/graphite/webapp/graphite/finders/utils.py", line 140, in fetch
    now=now, requestContext=requestContext

  File "/var/lib/graphite/webapp/graphite/node.py", line 30, in fetch
    result = self.reader.fetch(startTime, endTime, now, requestContext)

  File "/var/lib/graphite/webapp/graphite/readers/whisper.py", line 74, in fetch
    self.real_metric_path, start, step, values, aggregation_method, self.get_raw_step())

  File "/var/lib/graphite/webapp/graphite/readers/utils.py", line 116, in merge_with_carbonlink
    func=aggregation_method, raw_step=raw_step)

  File "/var/lib/graphite/webapp/graphite/readers/utils.py", line 72, in merge_with_cache
    consolidated = [(i, consolidate(func, consolidated_dict[i])) for i in consolidated_dict]

  File "/var/lib/graphite/webapp/graphite/readers/utils.py", line 53, in consolidate
    raise Exception("Invalid consolidation function: '%s'" % func)

Exception: Invalid consolidation function: 'avg_zero'
```

The query succeedes most of the times.

My storage-aggregation.conf:

```[all]
pattern = .*
xFilesFactor = 0.0
aggregationMethod = avg_zero
```

This is the output of whisper-info on one of the files:

```maxRetention: 63072000
xFilesFactor: 0.0
aggregationMethod: avg_zero
fileSize: 250720

Archive 0
retention: 1209600
secondsPerPoint: 60
points: 20160
size: 241920
offset: 40

Archive 1
retention: 63072000
secondsPerPoint: 86400
points: 730
size: 8760
offset: 241960
```

I believe this function should also include avg_zero, function which is available for carbon daemon.

https://github.com/graphite-project/graphite-web/blob/824800a4218c8cb9ab56ca67277b0f7774a7ff1d/webapp/graphite/readers/utils.py#L39-L53

**Comment by DanCech** [comment]
Yes, that makes sense.  Shouldn't be too hard to implement

**Comment by eachirei** [comment]
Thanks! Do you have an ETA for the fix?

**Comment by DanCech** [comment]
Someone will need to find time to work on it, I'd be happy to review a PR

**Comment by eachirei** [comment]
This is my attempt at fixing the issue:  https://github.com/graphite-project/graphite-web/pull/2338

**Comment by deniszh** [comment]
Merged to master, backported to 1.1.x, will be released in 1.1.5
